src/routes/search-route.jsx

```diff
@@ -1,3 +1,413 @@
+import { useQuery } from "@apollo/client";
+import Fuse from "fuse.js";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import Select from "react-select";
+import { Header, MovieCardList } from "../components";
+import ControlledCheckbox from "../components/form/advanced-form-inputs/controlledCheckbox";
+import Slider from "../components/form/advanced-form-inputs/slider";
+import TextInput from "../components/form/advanced-form-inputs/textInput";
+import { GET_KEYWORDS, GET_MOVIES } from "../graphql";
+import useDebounce from "../hooks/use-debounce";
+
+const decades = [
+  { label: "1970s", value: "1970" },
+  { label: "1980s", value: "1980" },
+  { label: "1990s", value: "1990" },
+  { label: "2000s", value: "2000" },
+  { label: "2010s", value: "2010" },
+  { label: "2020s", value: "2020" },
+];
+const sortingOptions = [
+  { label: "Title", value: "title" },
+  { label: "Popularity", value: "popularity" },
+  { label: "Recently Added", value: "recentlyAdded" },
+];
+
+/**
+ * Search route page
+ */
 export function SearchRoute() {
-  return <>{/* <Browse /> */}This is the search page</>;
+  const criteriaInit = {
+    sortingCriteria: "title",
+    rtMin: "20",
+    rtMax: "90",
+    inclNoRt: false,
+  };
+  const [criteria, setCriteria] = useState(criteriaInit);
+
+  const { register, control, watch, handleSubmit } = useForm({
+    defaultValues: criteriaInit,
+  });
+  const { data: keywordsData } = useQuery(GET_KEYWORDS);
+
+  const keywords = [];
+  for (let i = 0; i < keywordsData?.keywords.length; i++) {
+    const obj = {};
+    obj["label"] = keywordsData?.keywords[i].name;
+    obj["value"] = keywordsData?.keywords[i].id;
+    keywords.push(obj);
+  }
+
+  const onSubmit = useDebounce(setCriteria, 300);
+
+  return (
+    <>
+      <Header />
+      <main className="">
+        <form onChange={handleSubmit(onSubmit)}>
+          <div className="flex justify-between gap-4 mt-12 w-4/5 mx-auto">
+            <div className="w-1/3 flex flex-col gap-4 justify-start">
+              <span className="font-bold text-xl">Keywords</span>
+              <Controller
+                control={control}
+                defaultValue={[]}
+                name="keywords"
+                render={({ field: { onChange, value, ref } }) => {
+                  const val = keywords.filter((c) => value.includes(c.value));
+                  return (
+                    <Select
+                      inputRef={ref}
+                      value={val}
+                      styles={{
+                        control: (baseStyles, state) => ({
+                          ...baseStyles,
+                          borderColor: state.isFocused ? "black" : "black",
```

Comment:
No need for ternary here if both sides of it resolve to the same thing. You have this a few times throughout the code where this can be fixed.